trunk/SUAVE/Components/Energy/Converters/Supersonic_Nozzle.py

```diff
@@ -169,7 +175,156 @@ def compute(self,conditions):
         self.outputs.velocity                = u_out
         self.outputs.static_pressure         = P_out
         self.outputs.area_ratio              = area_ratio
-            
-    
+        
+    def compute_limited_geometry(self,conditions):
+        
+        """This is a variable geometry nozzle component that allows 
+        for supersonic outflow. all possible nozzle conditions, including 
+        overexpansion and underexpansion.
+        
+        Assumptions:
+        Constant polytropic efficiency and pressure ratio
+        
+        Source:
+        https://web.stanford.edu/~cantwell/AA283_Course_Material/AA283_Course_Notes/
+        https://web.stanford.edu/~cantwell/AA210A_Course_Material/AA210A_Course_Notes/
+        
+        Inputs:
+        conditions.freestream.
+          isentropic_expansion_factor         [-]
+          specific_heat_at_constant_pressure  [J/(kg K)]
+          pressure                            [Pa]
+          stagnation_pressure                 [Pa]
+          stagnation_temperature              [K]
+          gas_specific_constant               [J/(kg K)] 
+          mach_number                         [-]
+        self.inputs.
+          stagnation_temperature              [K]
+          stagnation_pressure                 [Pa]
+                   
+        Outputs:
+        self.outputs.
+          stagnation_temperature              [K]  
+          stagnation_pressure                 [Pa]
+          stagnation_enthalpy                 [J/kg]
+          mach_number                         [-]
+          static_temperature                  [K]
+          static_enthalpy                     [J/kg]
+          velocity                            [m/s]
+          static_pressure                     [Pa]
+                
+        Properties Used:
+        self.
+          pressure_ratio                      [-]
+          polytropic_efficiency               [-]
+          max_area_ratio                      [-]
+          min_area_ratio                      [-]
+        """           
+        
+        #unpack the values
+        
+        #unpack from conditions
+        gamma    = conditions.freestream.isentropic_expansion_factor
+        Cp       = conditions.freestream.specific_heat_at_constant_pressure
+        Po       = conditions.freestream.pressure
+        Pto      = conditions.freestream.stagnation_pressure
+        Tto      = conditions.freestream.stagnation_temperature
+        R        = conditions.freestream.gas_specific_constant
+        Mo       = conditions.freestream.mach_number
+        To       = conditions.freestream.temperature
+        
+        #unpack from inputs
+        Tt_in    = self.inputs.stagnation_temperature       #2400?  Why not 686 due to ray?
```

Comment:
Comments?

Reply:
Input stagnation temperature is still the maximum temperature given thermal limitations of the material in question, which, for nozzles, could sit at around 2400 K.

Separately, Rayleigh code computes the minimum stagnation temperature necessary to achieve thermal choking depending on freestream Mach number and other properties.

The program then selects the lowest of these 2 temperatures  to work with and passes its value to the nozzle script.